langchain4j/src/main/java/dev/langchain4j/service/tool/ToolHallucinationStrategy.java

```diff
@@ -0,0 +1,25 @@
+package dev.langchain4j.service.tool;
+
+import static dev.langchain4j.internal.Exceptions.runtime;
+
+import dev.langchain4j.agent.tool.ToolExecutionRequest;
+import java.util.function.Function;
+
+public enum ToolHallucinationStrategy implements Function<ToolExecutionRequest, String> {
+    THROW_EXCEPTION,
+    ERROR_MESSAGE;
+
+    public String apply(ToolExecutionRequest toolExecutionRequest) {
+        switch (this) {
+            case THROW_EXCEPTION -> {
+                throw runtime(
+                        "Something is wrong, the tool %s was called but it is not a part of the available tools",
+                        toolExecutionRequest.name());
+            }
+            case ERROR_MESSAGE -> {
```

Comment:
Remove this for now, since this is not a tested behavior and we don't want to provide it out-of-the-box.